infra/terraform/modules/user_nfs_softnas/ec2.tf

```diff
@@ -0,0 +1,142 @@
+data "aws_ami" "softnas" {
+  most_recent = true
+
+  filter {
+    name = "name"
+    values = ["SoftNAS Cloud Meter*"]
+  }
+
+  filter {
+    name = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name = "owner-alias"
+    values = ["aws-marketplace"]
+  }
+
+  # SoftNAS account ID
+  owners = ["679593333241"]
+}
+
+
+resource "aws_key_pair" "softnas" {
+  key_name   = "${var.env}-softnas"
+  public_key = "${var.ssh_public_key}"
+}
+
+resource "aws_security_group" "softnas" {
+  name        = "${var.env}-softnas"
+  description = "Allow NFS from cluster and HTTP from SSH bastions"
+  vpc_id = "${var.vpc_id}"
+
+  ingress {
+    from_port   = 2049
+    to_port     = 2049
+    protocol    = "tcp"
+    security_groups = ["${var.node_security_group_id}"]
+  }
+
+  ingress {
+    from_port   = 2049
+    to_port     = 2049
+    protocol    = "udp"
+    security_groups = ["${var.node_security_group_id}"]
+  }
+
+  ingress {
+    from_port = 443
+    to_port = 443
+    protocol = "tcp"
+    security_groups = ["${var.bastion_security_group_id}"]
+  }
+
+  ingress {
+    from_port = 22
+    to_port = 22
+    protocol = "tcp"
+    security_groups = ["${var.bastion_security_group_id}"]
+  }
+
+  ingress {
+    from_port = 8
+    to_port = 0
+    protocol = "icmp"
+    security_groups = ["${var.bastion_security_group_id}"]
+  }
+
+  ingress {
```

Comment:
That allows the softnas instances to communicate with each other via any protocol on any port - they communicate via SSH, FTP, NFS and ICMP to handle replication and HA failover.